xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp

```diff
@@ -33,15 +33,17 @@ void CRendererDXVA::GetWeight(std::map<RenderMethod, int>& weights, const VideoP
   const AVPixelFormat av_pixel_format = picture.videoBuffer->GetFormat();
   const DXGI_FORMAT dxgi_format = GetDXGIFormat(av_pixel_format, __super::GetDXGIFormat(picture));
 
-  const bool systemUsesHDR =
-      DX::Windowing()->IsHDROutput() || DX::Windowing()->IsHDRDisplaySettingEnabled();
-
   CEnumeratorHD enumerator;
   enumerator.Open(picture.iWidth, picture.iHeight, dxgi_format);
 
   if (av_pixel_format == AV_PIX_FMT_D3D11VA_VLD)
   {
-    if (enumerator.SupportedConversions(SupportedConversionsArgs{picture, systemUsesHDR}).empty())
+    const bool renderAsHDR =
+        RenderAsHDR(static_cast<AVColorPrimaries>(picture.color_primaries),
```

Comment:
Seems this code block can be moved before  `if (av_pixel_format == AV_PIX_FMT_D3D11VA_VLD)`  to a avoid repeat the same code in `else` branch (only `RenderAsHDR` or even all block until  return `"DXVA will not be used."`)

Reply:
I split the original `if (enumerator.SupportedConversions())` lines for readability, the compiler should optimize that stuff away.
I suppose the reason there were 2 instead of just the one was for the few tests in the `av_pixel_format != AV_PIX_FMT_D3D11VA_VLD` case where flow doesn't get to `SupportedConversions`: `if (dxgi_format == DXGI_FORMAT_UNKNOWN)` and incompatible texture format.

I'll suggest a refactor in a separate commit.